### DISCUSSION TITLE
U and V grids in awp130pgrb files on AWS

### DISCUSSION THREAD

**Discussion by csteele2** [NOAA-GSL/GSL-Regional-Model-Forum#40] [Q&A] [answered]
cfgrib is fairly finicky anyway, but while investigating trying to decode files I have downloaded, I noticed something peculiar about the index files for the U and V winds. This is one of the awp130pgrb idx files I grabbed off [AWS](https://registry.opendata.aws/noaa-rap/).

    210:11341265:d=2023051201:MASSDEN:8 m above ground:anl:
    211.1:11412922:d=2023051201:UGRD:10 m above ground:anl:
    211.2:11412922:d=2023051201:VGRD:10 m above ground:anl:
    212:11543958:d=2023051201:PRATE:surface:anl:

I noticed a couple things, the message IDs for U and V have decimals, and according to the second column, both messages are 0 bytes. This wind fields are rather important to me, so is there any inkling on whether these are being encoded correctly, and if so, how to go about decoding them?

**Comment by blaylockbk** [comment, marked as answer]
I don't understand everything about this, but the grib2 standard allows packing multiple fields into a single message, known as "submessage." In your example, the U and V fields are packed together in the same message (211).

See here for more: https://www.cpc.ncep.noaa.gov/products/wesley/wgrib2/submessages.html

**Reply by csteele2** [reply]
Okay, so I was taking a page out of your book. Since I do these things in large volumes, I don't download the entire file, but a subset based on that second column. So whenever I am doing a U grid, I just have to look at the next two lines then, and V grid, the previous two. IOW, download them as one set that will have two vars. Okay, thanks! That seems to have addressed my problem!
